client/components/tinymce/plugins/simple-payments/dialog/index.jsx

```diff
@@ -414,6 +414,13 @@ export default connect( ( state, { siteId } ) => {
 		siteId = getSelectedSiteId( state );
 	}
 
+	const productFormValues = getProductFormValues( state );
+	let isProductImageBeingUploaded = false;
+
+	if ( productFormValues ) {
+		isProductImageBeingUploaded = productFormValues.featuredImageId === 'uploading';
+	}
```

Comment:
What if we made this a part of the form validation? Let the validator function check if `featuredImageId === 'uploading'` and set a validation error if it is (`errors.featuredImageId = 'uploading'`). Simple and straighforward.